scripts/launch-s0-instance.sh

```diff
@@ -3,81 +3,233 @@
 me=$(basename $0)
 dir=$(readlink -f $(dirname $0))
 elastio_aws_id=537513441174
-security_group="elastio-s0-server"
+default_security_group="elastio-s0-server"
 default_instance_type="m5d.2xlarge"
 default_instance_name="$(whoami)-s0-server"
+default_block_size=fixed
+bootstrap=/tmp/s0-bootstrap.yml
+db_mount=/mnt/elastio
+unit_file=/etc/systemd/user/s0.service
+service=s0
+
+create_s0_bootstrap()
+{
+    rm -f $bootstrap
+    cat << EOF > $bootstrap
+#cloud-config
+# Bootstrap a 's0' server on Amazon Linux 2.
+# This script is intended to run as part of the cloud init
+
+write_files:
+  - path: /home/ec2-user/.aws/config
+    owner: ec2-user:ec2-user
+    content: |
+      [default]
+      output = json
+
+  - path: $UNIT_FILE
+    owner: ec2-user:ec2-user
+    permissions: '0664'
+    content: |
+      [Unit]
+      Description="Elastio Scale-to-Zero service"
+      After=network.target
+
+      [Service]
+      User=ec2-user
+      StartLimitInterval=5
+      StartLimitBurst=10
+      TimeoutStopSec=5min
+      ExecStart=/bin/bash -c '/usr/bin/s0 --log-output compact vault serve s3 --bucket $S0_BUCKET --key-id $S0_KMS $BLOCK_SIZE --server-address 0.0.0.0 --db-path $DB_MOUNT 2>&1 | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" >> /var/log/s0-service.log'
+      Restart=always
+      RestartSec=5
+
+      [Install]
+      WantedBy=multi-user.target
+
+runcmd:
+  # Format and mount nvme disk
+  - |
+    mkdir -p $DB_MOUNT
+    if [[ -b /dev/nvme1n1 && -b /dev/nvme2n1 ]]; then
+      # Two disks; make an array
+      mdadm --create --verbose /dev/md0 --level=0 --raid-devices=2 /dev/nvme1n1 /dev/nvme2n1
+      mkfs.xfs -f /dev/md0
+      mount /dev/md0 $DB_MOUNT
+    elif [ -b /dev/nvme1n1 ]; then
+      # One disk; no need to involve mdadm
+      parted -a optimal --script /dev/nvme1n1 mklabel gpt
+      parted -a optimal --script /dev/nvme1n1 mkpart primary 0% 100%
+      mkfs.xfs -f /dev/nvme1n1p1
+      mount /dev/nvme1n1p1 $DB_MOUNT
+    fi
+    chown -R ec2-user:ec2-user $DB_MOUNT
+
+  # Configure AWS region
+  - |
+    echo "region = \$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | grep -w region | cut -d: -f2 | tr -d '", ')" >> /home/ec2-user/.aws/config
+    cp -r /home/ec2-user/.aws /root/
```

Comment:
This shouldn't be necessary with `s0` since it doesn't ever need to run as `root`, but it will be necessary in a later automation that launches ingest VMs...

Reply:
Right now it's already necessary before the call `s0 vault create s3 ...` below. These cloud-init instructions in the `runcmd:` are running as `root`.

Reply:
That seems wrong.  We've discussed elsewhere why I don't think you should unconditionally `create` at all, but to do it as root...why?  You can `su` to run as the non-root user, although in this case I think the better fix is to not create the vault at all.

Reply:
See https://github.com/elastio/elastio-stack/pull/3#discussion_r560887316

Reply:
yes, I can `su`. But an instance is in some region despite the user you use inside of it. )) Why don't do this simplification and don't worry in the future that hypothetically some command can be executed as sudo or root and fail because we are decided to not do this simplification? ))